### PR TITLE
RevEng E2E testing

### LINKED ISSUE #2721: SQLite: Reverse Engineer
Implement all the things required to do
```ps1
Scaffold-DbContextTemplate EntityFramework.Sqlite
Scaffold-DbContext 'Data Source=Chinook.sqlite' EntityFramework.Sqlite
```

### PR DESCRIPTION
**Pull request by natemcmaster**
Move common E2E testing functionality into a common project.
Adds e2e testing for reverse engineering for SQLite.

Fix #2721

### CODE REVIEW

**Comment by natemcmaster** [comment]
cc: @lajones

File: test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void It_generates_for_simple_db()
+        {
+            var testStore = SqliteTestStore.GetOrCreateShared("SimpleReveng", ()=> {} );
+            testStore.ExecuteNonQuery(@"CREATE TABLE IF NOT EXISTS parent ( Id INT PRIMARY KEY);
+CREATE TABLE IF NOT EXISTS child (Id INT PRIMARY KEY, ParentId INT, FOREIGN KEY (ParentId) REFERENCES parent (Id ));
```
Comment (lajones) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs]: Nit: extra space after Id near end of line.

File: test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/SqlServerE2ETests.cs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                NamedReferences =
+                    {
+                        "EntityFramework.Core",
+                        "EntityFramework.Relational",
+                        "EntityFramework.SqlServer",
+#if DNXCORE50 || NETCORE50
+                   "System.Data.Common",
+                   "System.Linq.Expressions",
+                   "System.Reflection",
+    #else
```
Comment (lajones) [inline test/EntityFramework.SqlServer.Design.FunctionalTests/ReverseEngineering/SqlServerE2ETests.cs]: Nit: could you fix the spacing on the 3 lines above so they match the EntityFramework.* lines above?

File: test/EntityFramework.Sqlite.Design.FunctionalTests/App.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="System.Collections.Immutable" publicKeyToken="b03f5f7f11d50a3a"/>
+        <bindingRedirect oldVersion="1.1.36.0" newVersion="1.1.37.0"/>
+      </dependentAssembly>
```
Comment (lajones) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/App.config]: Nit: Maybe add a comment saying why this is necessary / when we can remove it.

Comment (natemcmaster) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/App.config]: Is it only so TestDriven.NET works? Or are there other reasons?

File: test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+        protected override E2ECompiler GetCompiler() => new E2ECompiler
+            {
+                NamedReferences =
+                    {
+                        "EntityFramework.Core",
+                        "EntityFramework.Relational",
+                        "EntityFramework.Sqlite",
+                        "Microsoft.Data.Sqlite",
+                        #if DNXCORE50
```
Comment (lajones) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs]: Nit: usually have #if's at beginning of line.

File: test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+{
+    public class SqliteE2ETest : E2ETestBase
+    {
+        public SqliteE2ETest(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void It_generates_for_simple_db()
```
Comment (lajones) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/ReverseEngineering/SqliteE2ETest.cs]: This is good for now. Make sure to log yourself an issue to a) extend this to more complicated database and b) implement the other 2 tests (the custom templates tests) as in SqlServerE2ETests.

**Comment by lajones** [comment]
:shipit:

File: test/EntityFramework.Relational.Design.FunctionalTests/EntityFramework.Relational.Design.FunctionalTests.xproj [hunk 6]
```diff
@@ -0,0 +1,22 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="__ToolsVersion__" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">14.0</VisualStudioVersion>
+    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
+  </PropertyGroup>
+  <Import Project="$(VSToolsPath)\DNX\Microsoft.DNX.Props" Condition="'$(VSToolsPath)' != ''" />
+  <PropertyGroup Label="Globals">
+    <ProjectGuid>06e77a73-3b5b-46f0-b366-e4b3f32ac60c</ProjectGuid>
```
Comment (bricelam) [inline test/EntityFramework.Relational.Design.FunctionalTests/EntityFramework.Relational.Design.FunctionalTests.xproj]: Don't reuse GIUDs :smile: (I'll fix in my next PR)

File: test/EntityFramework.Sqlite.Design.FunctionalTests/EntityFramework.Sqlite.Design.FunctionalTests.csproj [hunk 7]
```diff
@@ -34,9 +34,18 @@
     <Folder Include="Properties\" />
   </ItemGroup>
   <ItemGroup>
+    <None Include="App.config" />
     <None Include="project.json" />
+    <None Include="ReverseEngineering\Expected\child.expected" />
```
Comment (bricelam) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/EntityFramework.Sqlite.Design.FunctionalTests.csproj]:
```XML
<CopyToOutputDirectory>Always</CopyToOutputDirectory>
```
(I'll fix in my next PR)

Comment (natemcmaster) [inline test/EntityFramework.Sqlite.Design.FunctionalTests/EntityFramework.Sqlite.Design.FunctionalTests.csproj]: I caught this one in #2833

**Pull request merged**